index.php

```diff
@@ -27,6 +27,121 @@ function gutenberg_menu() {
 }
 add_action( 'admin_menu', 'gutenberg_menu' );
 
+
+$registered_blocks = array();
+
+/**
+ * Registers a block.
+ *
+ * @param  string $slug Block slug including namespace.
+ * @param  array  $settings Block settings.
+
+ * @return array            The block, if it has been successfully registered.
+ */
+function register_block( $slug, $settings ) {
+	global $registered_blocks;
+
+	if ( ! is_string( $slug ) ) {
+		$message = __( 'Block slugs must be strings.' );
+		_doing_it_wrong( __FUNCTION__, $message, '0.1.0' );
+		return;
+	}
+
+	$slug_matcher = '/^[a-z0-9-]+\/[a-z0-9-]+$/';
+	if ( ! preg_match( $slug_matcher, $slug ) ) {
+		$message = __( 'Block slugs must contain a namespace prefix. Example: my-plugin/my-custom-block' );
+		_doing_it_wrong( __FUNCTION__, $message, '0.1.0' );
+		return;
+	}
+
+	if ( isset( $registered_blocks[ $slug ] ) ) {
+		return;
+	}
+
+	$settings['slug'] = $slug;
+	$registered_blocks[ $slug ] = $settings;
+
+	return $settings;
+}
+
+/**
+ * Unregisters a block.
+ *
+ * @param  string $slug Block slug.
+ * @return array        The previous block value, if it has been
+ *                        successfully unregistered; otherwise `null`.
+ */
+function unregister_block( $slug ) {
+	global $registered_blocks;
+	if ( ! isset( $registered_blocks[ $slug ] ) ) {
```

Comment:
Why not trigger a `_doing_it_wrong` here if this is set, like we do on the client side?